aten/src/THC/generic/THCTensorTopK.cu

```diff
@@ -19,6 +19,9 @@ void THCTensor_(topk)(THCState* state,
 
   THArgCheck(dim >= 0 && dim < numDims, 6, "dim not in range");
 
+  if (k == 0) {
```

Comment:
you should bail out after you've correctly resized topK and indices (line 37), otherwise the size of your outputs will be incorrect (see e.g. cpu output size in this situation):
```
In [2]: a=torch.randn(10,5)                                                                                                                                                    

In [4]: out = a.topk(0, dim=1)                                                                                                                                                 

In [5]: out[0].size()                                                                                                                                                          
Out[5]: torch.Size([10, 0])
```